site/content/blog/2021-08-16-zap-fileupload-addon/index.md

```diff
@@ -0,0 +1,50 @@
+
+## Configuration
+File upload functionality generally has 2 endpoints, one from where file is uploaded and one from where file is retrieved. It is necessary to know both these endpoints. While Active Scanning an application, file upload endpoint is already known but retrieval endpoint is not known to the scan rule hence there are configuration details specific to the retrieval endpoint.
+
+Under ZAP's Options dialog you will find a FileUpload section as shown below:
+
+![FileUpload](./images/fileupload-options-panel.png)
+ 
+### Explanation
+
+For finding the URL to retrieve/view the uploaded file, here are some options:
+
+1. In some applications, the URL to retrieve the uploaded file is **static** and doesn't change or only the file name is changed. For handling this type of configuration, options panel has **Static Location Configuration** where static URL is added into URI Regex field. URI Regex field also supports the dynamic file name by **${fileName}** parameter, for e.g. `http://<baseurl>/${fileName}`
+2. In some applications the URL to retrieve the uploaded file is present in the **file upload request's response**. For handling this type of configuration, options panel has **Parse Http Response Configuration** which has 2 parameters **Start Identifier** and **End Identifier**. These identifiers are used to locate the URL within the response.
+3. In some applications the URL to retrieve the uploaded file is present in the **response of a different URL** which is called a **preflight request**. E.g. Profile picture URL is part of profile page and hence we need to parse the response of the profile page to find the URL of the profile picture. For handling this type of configuration, the options panel has **Dynamic Location Configuration** which has a **URI Regex** and **Parse Http Response Configuration** which has **Start Identifier**, and **End Identifier**. So the File Upload add-on will invoke the URI mentioned in URI Regex and then parse the response using Start Identifier and End Identifier. URI Regex field also supports the dynamic file name by **${fileName}**
+
+As FileUpload add-on's scan rule will execute as part of Owasp ZAP Active Scan, so after filling these configuration details, run the active scan in order to find the vulnerabilities in FileUpload Functionality.
+
+## Vulnerability Coverage
+The FileUpload add-on's scan rule attempts to upload various types of files with payloads and then tries to retrieve them to identify vulnerabilities present in the file upload functionality. 
+
+FileUpload add-on uploads following types of files:
+- HTML and its variants
+- JSP and its variants
+- JPEG and GIF images
+- EICAR file
+- SVG images
+
+Vulnerabilities covered are:
+- Cross site scripting
+- Remote Code Execution by uploading JSP and its variants
+- No virus scan. 
+
+The FileUpload add-on is available in the marketplace and can be installed from within ZAP.
```

Comment:
It should be more clear that's only available for dev versions (weekly and built from source). The add-on was not yet released though.

Reply:
Done.